rules/import_middleman.bzl

```diff
@@ -297,10 +291,6 @@ def _file_collector_rule_impl(ctx):
                 ]),
             ),
         )
-    else:
-        dep_cc_infos = [dep[CcInfo] for dep in ctx.attr.deps if CcInfo in dep]
-        cc_info = cc_common.merge_cc_infos(cc_infos = dep_cc_infos)
```

Comment:
I think we should add back the else case here so that we dont `merge_cc_infos` and waste time in the Bazel 7 case

Reply:
You mean Bazel 6 right? Unless I misunderstood something :)

Reply:
merging is the Bazel 6 case, without the if/else we always merge and then throw away the result in Bazel 7

Reply:
I started with original` if else` clause by having more fields in the create_compilation_context, but then it kept failing on other places. 
the impl behind the merge_cc_info is also merging of the fields of each one, isn't that the same as the the new impl?

Reply:
ah i see `cc_info` is also returned below so this makes sense